.github/workflows/release-automation.yaml

```diff
@@ -42,68 +47,101 @@ jobs:
-          export release_version=${release_tag/v/}
-          if ["$STATIC_BINARY_VERSION" != "$release_version"] || ["$DYNAMIC_BINARY_VERSION" != "$release_version"]; then
-            echo "Version mismatch"
-            exit 1
+          if [ "${{ matrix.arch }}" = "amd64" ]; then
+              make TARGET_ARCH=${{ matrix.arch }} RELEASE_TAG=$RELEASE_TAG release
+          elif [ "${{ matrix.arch }}" = "arm64" ]; then
+              make CC=aarch64-linux-gnu-gcc TARGET_ARCH=${{ matrix.arch }} RELEASE_TAG=$RELEASE_TAG release
           fi
+      - name: Verify Release version
+        run: |
+            HOST_ARCH=''
+            case $(uname -m) in
+                x86_64) HOST_ARCH="amd64" ;;
+                aarch64) HOST_ARCH="arm64" ;;
+                *) echo "Error: Unsupported arch $(uname -m)"; exit 1 ;;
+            esac
+            if [ "${{ matrix.arch }}" = "$HOST_ARCH" ]; then
+              mkdir -p output/static output/dynamic
+              tar -xzf release/${{ env.DYNAMIC_BINARY_NAME }} -C ./output/dynamic
+              tar -xzf release/${{ env.STATIC_BINARY_NAME }} -C ./output/static
+              DYNAMIC_BINARY_VERSION=$(./output/dynamic/finch-daemon --version | grep -oP '\d+\.\d+\.\d+')
+              STATIC_BINARY_VERSION=$(./output/static/finch-daemon --version | grep -oP '\d+\.\d+\.\d+')
+              export release_tag=${{ env.RELEASE_TAG }}
+              export release_version=${release_tag/v/}
+              if [ "$STATIC_BINARY_VERSION" != "$release_version" ] || [ "$DYNAMIC_BINARY_VERSION" != "$release_version" ]; then
+                echo "Version mismatch"
+                exit 1
+              fi
+            else
+              echo "Architecture mismatch: Cross Arch validation is not supported. Skipping version verification"
+            fi
         shell: bash
       - uses: actions/upload-artifact@v4
         with:
-          name: artifacts
+          name: artifacts-${{ matrix.arch }}
           path: release/
           if-no-files-found: error
     outputs:
       release_tag: ${{ env.RELEASE_TAG }}
+      release_version: ${{ env.RELEASE_VERSION }}
       dynamic_binary_name: ${{ env.DYNAMIC_BINARY_NAME }}
       static_binary_name: ${{ env.STATIC_BINARY_NAME }}
+
   validate-artifacts:
     needs: generate-artifacts
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        arch: [amd64, arm64]
     steps:
       - uses: actions/checkout@v4
       - uses: actions/download-artifact@v4
         with:
-          name: artifacts
+          name: artifacts-${{ matrix.arch }}
           path: release/
       - run: bash scripts/verify-release-artifacts.sh ${{ needs.generate-artifacts.outputs.release_tag }}
+        env:
+          TARGET_ARCH: ${{ matrix.arch }}
   create-release:
     needs: [generate-artifacts, validate-artifacts]
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
Do we really need to run this step once per arch? Can't we just run it once since its just uploading files?

Reply:
this can be optimized

Reply:
Optimized it.